bela/render.cpp

```diff
@@ -0,0 +1,263 @@
+    }
+    else
+    {
+        rt_printf("Decoding failed\n");
+    }
+
+}
+
+void on_state_changed_callback(void *ptr, chirp_connect_state_t old_state, chirp_connect_state_t new_state)
+{
+    rt_printf("State changed\n");
+}
+
+void on_receiving_callback(void *ptr, uint8_t *payload, size_t length, uint8_t channel)
+{
+    rt_printf("Receiving data\n");
+}
+
+void on_sent_callback(void *ptr, uint8_t *payload, size_t length, uint8_t channel)
+{
+    rt_printf("Data sent\n");
+}
+
+void on_sending_callback(void *ptr, uint8_t *payload, size_t length, uint8_t channel)
+{
+    rt_printf("Sending data\n");
+}
+
+bool setup(BelaContext *context, void *userData)
+{
+    /*
+     * Initialise global variables and buffers.
+     */
+    input_channel = 0;
+    block_size = context->audioFrames;
+    buffer_size = block_size * 8;
+    input_buffer = (float*) calloc(buffer_size, sizeof(float));
+    output_buffer = (float*) calloc(buffer_size, sizeof(float));
+
+
+    connect = new_chirp_connect(CHIRP_APP_KEY, CHIRP_APP_SECRET);
+    if (connect == NULL)
+    {
+        rt_printf("Connect is NULL\n");
+    }
+
+    chirp_connect_error_code_t ret = chirp_connect_set_config(connect, CHIRP_APP_CONFIG);
+    if(ret != CHIRP_CONNECT_OK)
+    {
+        rt_printf("Set config error\n");
+    }
+
+    chirp_connect_callback_set_t callbacks = {0};
+    callbacks.on_state_changed = on_state_changed_callback;
+    callbacks.on_sending = on_sending_callback;
+    callbacks.on_sent = on_sent_callback;
+    callbacks.on_receiving = on_receiving_callback;
+    callbacks.on_received = on_received_callback;
+    ret = chirp_connect_set_callbacks(connect, callbacks);
+    if(ret != CHIRP_CONNECT_OK)
+    {
+        rt_printf("Callback error\n");
+    }
+
+    ret = chirp_connect_set_callback_ptr(connect, connect);
+    if(ret != CHIRP_CONNECT_OK)
+    {
+        rt_printf("Callbackptr error\n");
+    }
+
+    /*
+     * Tell the SDK it can decode what it sends.
+     */
+    ret = chirp_connect_set_auto_mute(connect, false);
+    if(ret != CHIRP_CONNECT_OK)
+    {
+        rt_printf("Auto mute error\n");
+    }
+
+    ret = chirp_connect_set_volume(connect, 0.9);
+    if(ret != CHIRP_CONNECT_OK)
+    {
+        rt_printf("Set volume error\n");
+    }
+
+    ret = chirp_connect_set_input_sample_rate(connect, context->audioSampleRate);
+    if(ret != CHIRP_CONNECT_OK)
+    {
+        rt_printf("Set input sample rate error\n");
+    }
+
+    ret = chirp_connect_set_output_sample_rate(connect, context->audioSampleRate);
+    if(ret != CHIRP_CONNECT_OK)
+    {
+        rt_printf("Set output sample rate error\n");
+    }
+
+    ret = chirp_connect_start(connect);
+    if(ret != CHIRP_CONNECT_OK)
+    {
+        rt_printf("Start connect error\n");
+    }
+
+    /*
+     * Start the auxiliary tasks
+     */
+    if ((chirp_auxiliary_input_task = Bela_createAuxiliaryTask (&chirp_process_input_audio, 80, "chirp-auxiliary-input-task", connect)) == NULL) return false;
```

Comment:
Should the input task have the same priority as output?

Reply:
❓ They have the same priority

Reply:
Just curious, I would have thought that maybe the output processing should have a higher priority to avoid any audio glitches.

Reply:
It's okay, the environment is already made for optimised audio processing, and 80 of 99 is already a high enough priority, didn't have any glitches 😄